subsys/bluetooth/host/conn.c

```diff
@@ -2054,6 +2054,36 @@ int bt_conn_get_remote_info(struct bt_conn *conn,
 	}
 }
 
+int bt_conn_get_tx_power_level(struct bt_conn *conn, uint8_t type,
```

Comment:
I think we should consider "LE Enhanced Read Transmit Power Level command" when designing the API.
Maybe it would be best to move to a struct with input and output and call the enhanced version of the command if supported.
```
struct bt_conn_tx_power {
   uint8_t phy; /* input, values: 1M, 2M, CodedS2 CodedS8 */
   int8_t current_level; /* output */
   int8_t max_level; /* output */
}
```

We would then be able to use the same structure to read the remote TX power level as well.
If the HCI controller does not support the "LE Power Control Request" feature then we would call the BT_HCI_OP_READ_TX_POWER_LEVEL to read the max and current level.

Reply:
I see that the "LE Enhanced Read Transmit Power Level command" command also requires ``type`` as an input argument to read either max or current tx power. Both Tx Power commands (vanilla and enhanced) would need to be called twice to fill both output fields of the structure suggested by you.

However, I can add ``phy`` parameter to this API function that will determine whether an enhanced or vanilla (``phy = NULL``) version of Read Transmit Power Level command should be used:

``int bt_conn_get_tx_power_level(struct bt_conn *conn, uint8_t type, uint8_t *phy, int8_t *tx_power_level)``

``phy != NULL`` ->  "LE Enhanced Read Transmit Power Level command"
``phy == NULL`` ->  "Read Transmit Power Level command"

What do you think?

Reply:
Sorry, I confused the LE variant of the command with "Enhanced Read Transmit Power Level" command for BR/EDR in the previous comment.

I'm planning to change the API as you suggested.

Reply:
I addressed your comment in the last patch.